raft/src/main/java/org/apache/kafka/raft/internals/AddVoterHandler.java

```diff
@@ -156,6 +156,23 @@ public CompletableFuture<AddRaftVoterResponseData> handleAddVoterRequest(
             );
         }
 
+
+        if (!ackWhenCommitted && partitionState.hasSeenReplicaKey(voterKey)) {
+            // ackWhenCommitted is used to distinguish the request from admin
+            // or broker. If it comes from broker and had been in cluster, we should
+            // reject it.
```

Comment:
This is peculiar association. Why should "ack when committed" requests fail if the voter key is in the log?